montecristo/src/main/kotlin/com/datastax/montecristo/sections/configuration/CompactionSettings.kt

```diff
@@ -45,14 +46,20 @@ class CompactionSettings : DocumentSection {
         val args = super.createDocArgs(cluster)
         val compactionSection = StringBuilder()
 
-        val compactionThroughput = cluster.getSetting("compaction_throughput_mb_per_sec", ConfigSource.CASS, "16")
+        val compactionThroughput = if (cluster.databaseVersion is CassandraV41x) {
+            // xxx – 4.1+ clusters might still be using the legacy `compaction_throughput_mb_per_sec`
+            cluster.getSetting("compaction_throughput", ConfigSource.CASS, "64MiB/s")
```

Comment:
This is where the movement of the compaction throughput setting into the version class (like a number of the log strings) would make the section cleaner

Reply:
As above, we adopted the pattern of pushing this into the version classes to remove the complexity from the logic sections. I think that's still the pattern here, and no doubt other settings will over time fall into the same category and need new / old name support.

Reply:
As discussed, we have a path forward on how to handle this moving it to the versions and passing the method the yaml model object, eliminating the helpers on the model class which has no version knowledge and let the version class handle that. We can spot when a setting is not present to know if we need to drop down etc - the scale / scope of that work though can be a different ticket / PR.

Reply:
added in extra commit.
long comment added explaining what the intended longterm design to tackle this should be (as described over slack)